### PR TITLE
[bazel] fix dif unittests for autogen

### PR DESCRIPTION
**Pull request by drewmacrae**
updated bazel rules for new dependencies on dif_base.h and autogen
files.

Signed-off-by: Drew Macrae <drewmacrae@google.com>

### CODE REVIEW

File: sw/device/lib/dif/BUILD [hunk 1]
```diff
@@ -105,11 +105,10 @@ cc_library(
 cc_test(
     name = "uart_unittest",
     srcs = [
-        "autogen/dif_uart_autogen.inc",
+        "autogen/dif_uart_autogen.h",
```
Comment (cfrantz) [inline sw/device/lib/dif/BUILD:108]: Also: `autogen/dif_uart_autogen_unittest.cc` and similar for the other tests below.

File: sw/device/lib/dif/BUILD [hunk 2]
```diff
@@ -303,9 +302,11 @@ cc_library(
         "dif_hmac.c",
     ],
     hdrs = [
+        "autogen/dif_hmac_autogen.h",
         "dif_hmac.h",
     ],
     deps = [
+        "base",
```
Comment (cfrantz) [inline sw/device/lib/dif/BUILD:309]: I'm pretty sure this should be a label of the form `:base`.

File: sw/device/lib/dif/BUILD [hunk 3]
```diff
@@ -303,9 +302,11 @@ cc_library(
         "dif_hmac.c",
     ],
     hdrs = [
+        "autogen/dif_hmac_autogen.h",
```
Comment (cfrantz) [inline sw/device/lib/dif/BUILD:305]: I believe the autogen headers are considered private headers and should go in the `srcs` section.

**Review by timothytrippel — COMMENTED** [review]
LGTM, barring @cfrantz's comments. Thanks!

File: sw/device/lib/dif/BUILD [hunk 4]
```diff
@@ -300,12 +301,15 @@ cc_library(
...
         "dif_hmac.c",
     ],
     hdrs = [
         "dif_hmac.h",
     ],
     deps = [
+        ":base",
         "//hw/ip/hmac/data:hmac_regs",
         "//sw/device/lib/base",
     ],
```
Comment (timothytrippel) [inline sw/device/lib/dif/BUILD:315]: Autogen'd unit tests have been created and merged (#8304) into master so you could add a unit test target for HMAC too.

**Review by timothytrippel — APPROVED** [review]

File: sw/device/lib/dif/BUILD (hunk 4 above)
Comment (drewmacrae) [reply sw/device/lib/dif/BUILD:315]: I got that one working. Are there any others that are autogen'd I should be running?

**Review by moidx — APPROVED** [review]

**Pull request merged**